src/renderer/dx12.rs

```diff
@@ -511,18 +536,18 @@ impl RenderEngine {
                 None,
             );
         }
-
-        // Draw data loop.
-        let ctx = Rc::clone(&self.ctx);
-        let mut ctx = ctx.borrow_mut();
-        let ui = ctx.frame();
-        render_loop(ui);
-        let draw_data = ctx.render();
-
-        if let Err(e) = unsafe { self.render_draw_data(draw_data, idx) } {
-            eprintln!("{}", e);
-        };
-
+        if unsafe { SHOW_UI } {
```

Comment:
This is a kind of usage of `unsafe` that should be avoided **at all costs**.

We are operating in a concurrent environment, and statements like this **will** lead the compiler to take some unpredictable optimizations that lead to undefined behavior.

Reply:
removed